editor/src/components/editor/store/store-deep-equality-instances.ts

```diff
@@ -1215,28 +1274,25 @@ export function JSXElementChildKeepDeepEquality(): KeepDeepEqualityCall<JSXEleme
       return JSXFragmentKeepDeepEquality(oldElement, newElement)
     } else if (isJSXConditionalExpression(oldElement) && isJSXConditionalExpression(newElement)) {
       return JSXConditionalExpressionKeepDeepEquality(oldElement, newElement)
-    } else if (oldElement.type === 'ATTRIBUTE_VALUE' && newElement.type === 'ATTRIBUTE_VALUE') {
+    } else if (isJSExpressionValue(oldElement) && isJSExpressionValue(newElement)) {
       return JSXAttributeValueKeepDeepEqualityCall(oldElement, newElement)
     } else if (
-      oldElement.type === 'ATTRIBUTE_OTHER_JAVASCRIPT' &&
-      newElement.type === 'ATTRIBUTE_OTHER_JAVASCRIPT'
+      isJSExpressionOtherJavaScript(oldElement) &&
+      isJSExpressionOtherJavaScript(newElement)
     ) {
       return JSExpressionOtherJavaScriptKeepDeepEqualityCall()(oldElement, newElement)
-    } else if (
-      oldElement.type === 'ATTRIBUTE_NESTED_ARRAY' &&
-      newElement.type === 'ATTRIBUTE_NESTED_ARRAY'
-    ) {
+    } else if (isJSExpressionNestedArray(oldElement) && isJSExpressionNestedArray(newElement)) {
       return JSXAttributeNestedArrayKeepDeepEqualityCall()(oldElement, newElement)
-    } else if (
-      oldElement.type === 'ATTRIBUTE_NESTED_OBJECT' &&
-      newElement.type === 'ATTRIBUTE_NESTED_OBJECT'
-    ) {
+    } else if (isJSExpressionNestedObject(oldElement) && isJSExpressionNestedObject(newElement)) {
       return JSXAttributeNestedObjectKeepDeepEqualityCall()(oldElement, newElement)
-    } else if (
-      oldElement.type === 'ATTRIBUTE_FUNCTION_CALL' &&
-      newElement.type === 'ATTRIBUTE_FUNCTION_CALL'
-    ) {
+    } else if (isJSExpressionFunctionCall(oldElement) && isJSExpressionFunctionCall(newElement)) {
       return JSXAttributeFunctionCallKeepDeepEqualityCall()(oldElement, newElement)
+    } else if (isJSIdentifier(oldElement) && isJSIdentifier(newElement)) {
+      return JSIdentifierKeepDeepEquality()(oldElement, newElement)
+    } else if (isJSPropertyAccess(oldElement) && isJSPropertyAccess(newElement)) {
+      return JSPropertyAccessKeepDeepEquality()(oldElement, newElement)
+    } else if (isJSElementAccess(oldElement) && isJSElementAccess(newElement)) {
+      return JSElementAccessKeepDeepEquality()(oldElement, newElement)
     } else {
       return keepDeepEqualityResult(newElement, false)
```

Comment:
shouldn't we instead of this catch-all put an exhaustive check here so in the future we can't forget adding new deep equality results?

Reply:
Yeah, that would be better.

Reply:
Fixed in 857d3890a.